libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -1080,7 +1087,8 @@ void CuptiActivityProfiler::finalizeTrace(const Config& config, ActivityLogger&
   if (!process_name.empty()) {
     logger.handleDeviceInfo(
         {pid, process_name, "CPU"}, captureWindowStartTime_);
-    if (!cpuOnly_) {
+    if (!cpuOnly_ &&
+        !libkineto::hasPrivateUse1Type(config.selectedActivityTypes())) {
```

Comment:
@fwenguang I understand what you are trying to do here. When NVIDIA or AMD GPU or Private use1 is not present you don't want this to run. How about we update this to not be specific to private use 1?  
Like 
hasAcceleratorActivities() ? 
and this can check for GPU events, HIP (AMD) events , MTIA events and PrivateUse1 events?

Reply:
cc @aaronenyeshi wdyt?

Reply:
@briancoutinho I think this is okay for Florian's use case, but we should try to clean this up later.